spec/server/unit/user.spec.coffee

```diff
@@ -150,3 +178,124 @@ describe 'User', ->
         @user = new User({ coursePrepaid: undefined })
         expect(@user.prepaidIncludesCourse('course_2')).toBe(false)
         
+
+  describe '.updateMailChimp()', ->
+    beforeEach utils.wrap (done) ->
+      yield utils.clearModels([User])
+      done()
+      
+    mailChimp = require '../../../server/lib/mail-chimp'
+
+    it 'propagates user notification and name settings to MailChimp', utils.wrap (done) ->
+      user = yield utils.initUser({
+        emailVerified: true
+        firstName: 'First'
+        lastName: 'Last'
+        emails: {
+          diplomatNews: { enabled: true }
+          generalNews: { enabled: true }
+        }
+      })
+      spyOn(mailChimp.api, 'put').and.returnValue(Promise.resolve())
+      yield user.updateMailChimp()
+      expect(mailChimp.api.put.calls.count()).toBe(1)
+      args = mailChimp.api.put.calls.argsFor(0)
+      expect(args[0]).toMatch("^/lists/[0-9a-f]+/members/[0-9a-f]+$")
+      expect(args[1].email_address).toBe(user.get('email'))
+      diplomatInterest = _.find(mailChimp.interests, (interest) -> interest.property is 'diplomatNews')
+      announcementsInterest = _.find(mailChimp.interests, (interest) -> interest.property is 'generalNews')
+      for [key, value] in _.pairs(args[1].interests)
+        if key in [diplomatInterest.mailChimpId, announcementsInterest.mailChimpId]
+          expect(value).toBe(true)
+        else
+          expect(value).toBeFalsy()
+      expect(args[1].status).toBe('subscribed')
+      expect(args[1].merge_fields['FNAME']).toBe('First')
+      expect(args[1].merge_fields['LNAME']).toBe('Last')
+      user = yield User.findById(user.id)
+      expect(user.get('mailChimp').email).toBe(user.get('email'))
+      done()
+      
+    describe 'when user email is validated on MailChimp but not CodeCombat', ->
+      
+      it 'still updates their settings on MailChimp', utils.wrap (done) ->
+        email = 'some@email.com'
+        user = yield utils.initUser({
+          email
+          emailVerified: false
+          emails: {
+            diplomatNews: { enabled: true }
+          }
+          mailChimp: { email }
+        })
+        user = yield User.findById(user.id)
+        spyOn(mailChimp.api, 'get').and.returnValue(Promise.resolve({ status: 'subscribed' }))
+        spyOn(mailChimp.api, 'put').and.returnValue(Promise.resolve())
+        yield user.updateMailChimp()
+        expect(mailChimp.api.get.calls.count()).toBe(1)
+        expect(mailChimp.api.put.calls.count()).toBe(1)
+        args = mailChimp.api.put.calls.argsFor(0)
+        expect(args[1].status).toBe('subscribed')
```

Comment:
Changed a few of these to eg `args[1]?.status` since if the function just isn't called, it crashes the test suite